### ISSUE TITLE
Index.UpdateDocumentsAsync() got an exception about Reflection-based serialization has been disabled for this application

### ISSUE THREAD

**Issue by wooly905** [meilisearch/meilisearch-dotnet#521]
**Description**
When the app is set to "publish trimed" in csproj, System.Text.Json will disable reflection-based serialization.
``  <PublishTrimmed>true</PublishTrimmed>
``

In current meilisearch-dotnet source code, it uses System.Text.Json to serialize and deserialize. However, System.Text.Json doesn't support reflection-based serialization in above case. We will have an error message when UpdateDocumentsAsync() is invoked as shown in the screenshot below. 

All functions (for example, Index, Task, Document and so on) talk to meilisearch server will fail.

**Expected behavior**
UpdateDocumentsAsync() should work properly.

**Current behavior**
UpdateDocumentsAsync() throws InvalidOperationException as shown in the screenshot.

**Screenshots or Logs**
![image](https://github.com/meilisearch/meilisearch-dotnet/assets/18693681/2165d8e4-42a2-436f-bfbc-454ffab20b2f)

**Environment (please complete the following information):**
- OS: Windows 11 with .NET 8
- Meilisearch version: 1.6
- meilisearch-dotnet version: 0.14.7

**One possible solution: **
All functions talk to meilisearch api endpoints should provide new overloaded functions with a new parameter about JsonTypeInfo. So developers can decide what type to serialize and de-serialize as shown in the screenshot.

**Comment by curquiza** [comment]
@ahmednfwela, if you are around, is it something you experienced or managed to reproduce?

I'm not a Dotnet user, so it's not trivial for me

**Comment by ahmednfwela** [comment]
yes, this specific issue requires changing the API to allow for users to take control of serialization and de-serialization

I am working on multiple PRs in .net/dart/flutter and this is the top priority right now
